include/RE/A/Actor.cpp

```diff
@@ -496,6 +580,29 @@ namespace RE
 		return func(this, a_weapon, a_extraData, a_leftHand);
 	}
 
+	void Actor::VisitArmorAddon(TESObjectARMO* a_armor, TESObjectARMA* a_arma, std::function<void(bool a_firstPerson, RE::NiAVObject* a_obj)> a_visitor)
+	{
+		char addonString[WinAPI::MAX_PATH];
+		std::memset(addonString, 0, WinAPI::MAX_PATH);
+		a_arma->GetNodeName(addonString, this, a_armor, -1);
+
+		NiAVObject* skeletonRoot[2] = { Get3D(0), Get3D(1) };
+
+		if (skeletonRoot[1] == skeletonRoot[0]) {
+			skeletonRoot[1] = nullptr;
+		}
+
+		for (int32_t idx = 0; auto& root : skeletonRoot) {
+			if (root) {
+				auto obj = root->GetObjectByName(addonString);
+				if (obj) {
```

Comment:
If `obj` is guaranteed not null, then surely the callback should take it by reference?